### PR TITLE
Passing `SAMPLE_PLUGIN_OPTS` if already set

### PR DESCRIPTION
**Pull request by alecharp**
Follow up of #2940, revert in #2951.

### Testing done

I ran 

```
$ PLUGINS=prism-api TEST=io.jenkins.plugins.prism.SourceDirectoryFilterTest bash local-test.sh
[...]
[INFO] Running InjectedTest
Feb 28, 2024 5:26:41 PM org.jvnet.hudson.test.WarExploder explode
INFO: Exploding /Users/alecharpentier/.m2/repository/org/jenkins-ci/main/jenkins-war/2.447/jenkins-war-2.447.war into /Users/alecharpentier/workspaces/jenkinsci/bom/sample-plugin/target/jenkins-for-test
[...]
```

and saw that the current weekly version was used for the tests.

I ran

```
$ PLUGINS=prism-api TEST=io.jenkins.plugins.prism.SourceDirectoryFilterTest SAMPLE_PLUGIN_OPTS="-Djenkins.version=2.445" bash local-test.sh
[...]
[INFO] Running InjectedTest
Feb 28, 2024 5:29:58 PM org.jvnet.hudson.test.WarExploder explode
INFO: Exploding /Users/alecharpentier/.m2/repository/org/jenkins-ci/main/jenkins-war/2.445/jenkins-war-2.445.war into /Users/alecharpentier/workspaces/jenkinsci/bom/sample-plugin/target/jenkins-for-test
[...]
```

and saw that it used Jenkins 2.445 as specified in the `SAMPLE_PLUGIN_OPTS` variable.
I also validated that the execution worked correctly in a git-bisect on the core.

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by basil — COMMENTED** [review]
This doesn't match the existing formatting of this file which uses tab-based indentation rather than spaces.

**Comment by alecharp** [comment]
You're absolutely right. Sorry for that.

**Review by basil — COMMENTED** [review]
Thinking about this some more, I'm not sure it makes sense to append `-DInjectedTest` to `SAMPLE_PLUGIN_OPTS` if it is already set. That seems like it would violate the principle of least surprise by changing a value the user has provided. Perhaps we should get rid of the `if` and keep only the `else` (that is, only setting `SAMPLE_PLUGIN_OPTS` if it is not already set) by inverting the `if` (from `-n` to `-z`).

**Comment by basil** [comment]
The problem is that if the user sets `SAMPLE_PLUGIN_OPTS` to `-Dtest=[SomeOtherTest]`, it doesn't make sense to append `-Dtest=InjectedTest`. If we are going to allow external input for `SAMPLE_PLUGIN_OPTS` that we can't control, we can't make any assumptions about it. Therefore we can't assume that it's safe to append `-Dtest`. So the choices are:

- Detect if the user passed in `-Dtest` in `SAMPLE_PLUGIN_OPTS` and do not append `-Dtest=InjectedTest` in this case (seems overly complicated), or
- Don't mess with `SAMPLE_PLUGIN_OPTS` if the user already passed it in. Make the user pass in `-Dtest=InjectedTest` as well as any custom arguments

I think the second choice is simpler.

File: local-test.sh [hunk 1]
```diff
@@ -9,7 +9,9 @@ cd "$(dirname "$0")"
 LATEST_LINE=weekly
 : "${LINE:=$LATEST_LINE}"
 
-SAMPLE_PLUGIN_OPTS=-Dtest=InjectedTest
+if [[ -z "${SAMPLE_PLUGIN_OPTS:-}" ]]; then
```
Comment (basil) [inline local-test.sh:12]: Can be simplified to

```suggestion
if [[ -z ${SAMPLE_PLUGIN_OPTS-} ]]; then
```

**Review by basil — APPROVED** [review]

**Pull request merged**